core/Tracker/VisitExcluded.php

```diff
@@ -180,13 +180,26 @@ private function isIpInRange()
         if ($cache->contains($key)) {
             $isInRanges = $cache->fetch($key);
         } else {
-            $isInRanges = $ip->isInRanges($this->getBotIpRanges());
+            if ($this->isChromeDataSaverUsed($ip)) {
+                $isInRanges = false;
+            } else {
+                $isInRanges = $ip->isInRanges($this->getBotIpRanges());
+            }
+
             $cache->save($key, $isInRanges);
         }
 
         return $isInRanges;
     }
 
+    private function isChromeDataSaverUsed(IP $ip)
+    {
+        // see https://github.com/piwik/piwik/issues/7733
+        return !empty($_SERVER['HTTP_VIA'])
+            && false !== strpos(strtolower($_SERVER['HTTP_VIA']), 'chrome-compression-proxy')
+            && $ip->isInRange('66.249.80.0/20');
```

Comment:
I don't know all googles IP ranges. Are they specified somewhere?  

I didn't use `getBotIpRanges()` as it might also contains non google ip addresses? And as it could be misused? Also `isInRange()` is very very slow (this made bulk tracking very slow, that's why it caches some parts). So tried to avoid too many calls.